service-tests/xml.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'xml', title: 'User defined XML source data' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('/Extension Name/$.addon.name/https://services.addons.mozilla.org/api/1.5/addon/indiegala-helper.json')
```

Comment:
Huh, this is an interesting idea: querying XML using jsonpath. Definitely clever!

Though I'm not sure it's that intuitive to a user. If a user were looking at some XML and the jsonpath documentation, would they know what to write?

My intuition is that we should use an xpath library, which would allow for less guesswork, more idiomatic queries, and probably more power. Though I'm open to being convinced.

Reply:
Definitely makes sense to use xpath for the xml.
I might just remove the xml from the current PR so it can be merged in a later PR as i don't really have much experience dealing with xml/xpath so would rather just get the json parts right first then the xml separately.